core/hex.py

```diff
@@ -0,0 +1,68 @@
+"""
+Contains Abalone-specific grid cell logic.
+"""
+
+from enum import Enum
+from lib.axial_hex import AxialHex
+from lib.lerp import lerp
+from constants import BOARD_MAXCOLS
+
+# neighbor cache to avoid neighborhood recalculation
+HEX_NEIGHBORS = {}
+
+class Hex(AxialHex):
+    """
+    A hex cell specific to the game of Abalone.
+    """
+
+    def __str__(self):
+        """
+        Returns the current cell in Abalone move notation, e.g. A1 etc.
+        :return: a string
+        """
+        return f"{chr(BOARD_MAXCOLS - self.y + ord('A') - 1)}{self.x + 1}"
+
+    def lerp(self, other, time):
+        """
+        Interpolates the cell between `self` and `other` at `time`%.
+        Used primarily for animation logic.
+        :param other: the ending cell to interpolate to
+        :param time: the time to interpolate to
+        :return: a Hex
+        """
+        return Hex(
+            x=lerp(self.x, other.x, time),
+            y=lerp(self.y, other.y, time),
+        )
+
+    def neighbors(self):
+        """
+        Finds all six neighbors of this cell.
+        :return: a list[Hex]
+        """
+        self_tuple = (self.x, self.y)
+        if self_tuple not in HEX_NEIGHBORS: # cache neighbors (ok with 61 cells)
+            HEX_NEIGHBORS[self_tuple] = [self.add(d.value) for d in HexDirection]
+        return HEX_NEIGHBORS[self_tuple]
+
+class HexDirection(Enum):
+    """
+    The collection of all six hex directions.
+    """
+
+    NW = Hex(0, -1)
+    NE = Hex(1, -1)
+    W = Hex(-1, 0)
+    E = Hex(1, 0)
+    SW = Hex(-1, 1)
+    SE = Hex(0, 1)
+
```

Comment:
just do `for direction in HexDirection` if you want to loop through all six directions for counting defenders and marble shapes, and use `Hex.add(cell, direction.value)` if you want to get the neighbor adjacent in that direction